airflow/security/permissions.py

```diff
@@ -66,14 +66,16 @@
 DAG_ACTIONS = {ACTION_CAN_READ, ACTION_CAN_EDIT, ACTION_CAN_DELETE}
 
 
-def resource_name_for_dag(dag_id):
+def resource_name_for_dag(dag_id, is_subdag=False):
     """Returns the resource name for a DAG id."""
     if dag_id == RESOURCE_DAG:
         return dag_id
 
     if dag_id.startswith(RESOURCE_DAG_PREFIX):
         return dag_id
 
-    # To account for SubDags
-    root_dag_id = dag_id.split(".")[0]
-    return f"{RESOURCE_DAG_PREFIX}{root_dag_id}"
+    if is_subdag:
+        # To account for SubDags
+        root_dag_id = dag_id.split(".", 1)[0]
```

Comment:
Is it possible to have a subdag of a dag with a dot in the name? eg `data.munge` with a subdag making it `data.munge.subdag`? If so, this won't return the right value for `root_dag_id`.

Reply:
Good point! And we actually know definitively what the root DAG ID is in this case (`root_dag_id`), so we should pass that in directly instead.

Reply:
We only know the `dag_id` and not the `root_dag_id`.

Reply:
`root_dag_id` is available on `DagModel`. Since we already query it for `is_subdag` we should be a ble to also get that value as well, I think?

Reply:
Oh, yeah!!

Reply:
Instead of `is_subdag` and `root_dag_id`, I'm using `parent_dag`. Only `subdags` have `parent_dag`. That solves it, i believe

Reply:
`parent_dag` is more expensive since it needs to load the entire DagModel object. Since this permission check is called a lot, I would prefer the check to be as light as possible. It would be better to use `root_dag_id` instead (which `parent_dag` is based on anyway!)

Reply:
Done